clients/java/src/test/java/io/camunda/client/OAuthCredentialsProviderTest.java

```diff
@@ -495,6 +566,45 @@ private void mockTokenRequest(final boolean withAssertion) {
     }
   }
 
+  private void mockTokenRequestWithResource(final boolean withAssertion, final String resource) {
```

Comment:
Could you please let `mockTokenRequest(final boolean withAssertion)` call this method here, similar to how you did it with the tests in the zeebe package? That reduces code duplication.